frappe/core/doctype/document_naming_rule/document_naming_rule.py

```diff
@@ -6,8 +6,19 @@
 import frappe
 from frappe.model.document import Document
 from frappe.utils.data import evaluate_filters
+from frappe import _
 
 class DocumentNamingRule(Document):
+	def validate(self):
+		self.validate_fields_in_conditions()
+
+	def validate_fields_in_conditions(self):
+		for condition in self.conditions:
+			docfields = frappe.get_meta(self.document_type).fields
```

Comment:
Don't think this will be changing on every iteration of `self.conditions`. Move this outside the for loop?